💥 challenge one (#1)
mohamadhammoud/forta-challenges: README.md

# Nethermind Forta Bot Deployment Agent

## Description

This agent detects specific transactions related to Nethermind bot deployments and updates on the Forta registry smart contract on the Polygon network. It monitors the `createAgent` and `updateAgent` methods from a specified deployer address.

## Supported Chains

- Polygon (Chain ID: 137)

> - List any other chains this agent can support e.g. Ethereum, BSC

Reviewer: Irrelevant lines autogenerated should be deleted.

## Alerts

- **FORTA-1 Create**

  - Fired when a transaction contains a call to the `createAgent` method from the specified deployer address
  - Severity is always set to "low" (mention any conditions where it could be something else)

>   - Type is always set to "info" (mention any conditions where it could be something else)

Reviewer: ```suggestion
  - Severity is always set to "low"
  - Type is always set to "info"
```
Similar suggestion regarding the autogenerated text.

  - Metadata includes `agentId`, `metadata`, and `chainIds`

- **FORTA-1 Update**
  - Fired when a transaction contains a call to the `updateAgent` method from the specified deployer address
  - Severity is always set to "low" (mention any conditions where it could be something else)
  - Type is always set to "info" (mention any conditions where it could be something else)

>   - Metadata includes `agentId`, `metadata`, and `chainIds`

Reviewer: Team common practice is to have the metadata components as sub-bullet points with a short description.

Reviewer: Applicable to both.

## Configuration

### forta.config.json

To configure the agent, create a `forta.config.json` file with the following content:

```json
{
  "agentId": "",
  "jsonRpcUrl": ""
}
```

### package.json

Ensure the package.json includes the Polygon chain ID:

```json
 "chainIds": [
    137
  ]

> ```

Reviewer: While appreciated, these lines aren't necessary. If the bot were more unique (i.e. configuration differed from standard bots significantly), then that would be more applicable. For _most_ bots, it's not really necessary.

Reviewer: Though, I get that things like things would be applicable in a lot of open source projects. 😎

> ## Usage

Reviewer: Similar suggestion to this section where you really only need to provide transaction with which to test the bot with and if multiple, could explain what activity each would be flagged.
